manticore/core/cpu/x86.py

```diff
@@ -4881,6 +4881,219 @@ def PCMPEQB(cpu, op0, op1):
             #    res = res | (0xff << i)
         op0.write(res)
 
+    ############################################################################
+    # Implementation of PCMPxSTRx Instructions:
+    # + PCMPESTRM, PCMPESTRI, PCMPISTRM, PCMPISTRI
+    ##
+    # '''
+    # See Ref: https://software.intel.com/sites/default/files/m/8/b/8/D9156103.pdf
+
+    # :param cpu: current CPU.
+    # :param op0: compare oprand 1
+    # :param op1: compare oprand 2
+    # :param op2: control byte
+    #             - op2[3:2] :
+    #               + 00: Equal Any
+    #               + 01: Ranges
+    #               + 10: Equal Each
+    #               + 11: Equal Ordered
+    # '''
+    #############################################################################
+    def _pcmpxstrx_srcdat_format(self, ctlbyte):
+        # Parse CTL Byte
+        ## Source Data Format
+        if ((Operators.EXTRACT(ctlbyte, 0, 2) & 1) == 0 ):
+            stepsize = 8
+        else:
+            stepsize = 16
+        return stepsize
+
+    def _pcmpxstri_output_selection(self, ctlbyte, res):
+        ## Output Selection
+        ###  PCMPESTRI/PCMPISTRI
+        stepsize = self._pcmpxstrx_srcdat_format(ctlbyte)
+        if (Operators.EXTRACT(ctlbyte, 6, 1) == 0):
+            oecx = 0
+            tres = res
+            while ((tres & 1) != 0):
+                oecx += 1
+                tres >>= 1
+            return oecx
+        else:
+            oecx = 0
+            tres = res
+            msbmask = (1 << (self.regfile._table['XMM0'].size/stepsize - 1))
```

Comment:
This is already great but I'll leave this comment here anyway.
Is self.regfile._table['XMM0'].size going to vary? If not, why not replace it with the actual constant ?
Other idea. Add regfile.sizeof?(self, reg): self.regfile._table[reg].size

Reply:
I agree changing self.regfile._table['XMM0'].size to actual constant. It's not going to change anyway.